library/src/main/java/com/wdullaer/materialdatetimepicker/date/DefaultDateRangeLimiter.java

```diff
@@ -203,7 +232,8 @@ private boolean isAfterMax(@NonNull Calendar calendar) {
     }
 
     @Override
-    public @NonNull Calendar setToNearestDate(@NonNull Calendar calendar) {
+    public @NonNull
+    Calendar setToNearestDate(@NonNull Calendar calendar) {
```

Comment:
setToNearest will need to take the disabled weekdays into account (you don't want to return a date that is disabled)